src/sidebar/components/ShareDialog/ExportAnnotations.tsx

```diff
@@ -126,10 +129,21 @@ function ExportAnnotations({
         selectedUser?.annotations ?? exportableAnnotations;
       const filename = `${customFilename ?? defaultFilename}.${format}`;
 
-      if (format === 'json') {
-        const exportData =
-          annotationsExporter.buildJSONExportContent(annotationsToExport);
-        downloadJSONFile(exportData, filename);
+      switch (format) {
+        case 'json': {
+          const exportData =
+            annotationsExporter.buildJSONExportContent(annotationsToExport);
+          downloadJSONFile(exportData, filename);
+          break;
+        }
+        case 'txt': {
+          const exportData = annotationsExporter.buildTextExportContent(
+            annotationsToExport,
+            group?.name,
+          );
+          downloadTextFile(exportData, filename);
+          break;
+        }
```

Comment:
Considering this `switch` will continue growing, I'm thinking the logic might be wrapped in the `AnnotationsExporter` itself, or a service which wraps calling the `AnnotationsExporter` appropriate method and function to download files.

That will simplify the `ExportAnnotations` test and let us test this conditional logic individually.

Reply:
I actually think this will be OK as is. The complicated parts here (generating the data and initiating the download) are already encapsulated, so I suspect that testing this with appropriate mocks won't be too hard. Have a stab at writing the tests and see.

Reply:
ok!